tests/common.go

```diff
@@ -1606,7 +1610,25 @@ func ScheduleApplications(testname string, errChan ...*chan error) []*scheduler.
 	Step("schedule applications", func() {
 		options := CreateScheduleOptions("", errChan...)
 		taskName := fmt.Sprintf("%s-%v", testname, Inst().InstanceID)
-		contexts, err = Inst().S.Schedule(taskName, options)
+		var specFactory *spec.Factory
```

Comment:
These changes are made to test ScheduleWithCustomAppSpecs. I will remove them before merging this PR.

Reply:
Testing ScheduleWithCustomAppSpecs code

```
        specFactory, err := spec.NewFactory(tests.Inst().SpecDir, tests.Inst().Provisioner, tests.Inst().S)
	if err != nil {
		return nil, controllers.ProcessError(err)
	}
	var appSpecs []*spec.AppSpec
	if len(c.AppKeys) > 0 {
		for _, key := range c.AppKeys {
			var appSpec *spec.AppSpec
			appSpec, err = specFactory.Get(key)
			if err != nil {
				return nil, controllers.ProcessError(err)
			}
			appSpecs = append(appSpecs, appSpec)
		}
	} else {
		appSpecs = specFactory.GetAll()
	}
	ScheduleWithCustomAppSpecs(appSpecs, ....)
```